src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts

```diff
@@ -270,14 +271,15 @@ export class NotebookService extends Disposable implements INotebookService {
 
 		let serializedContent: string;
 		if (contents) {
-			// Have to serialize contents again first, since our notebook code assumes input is based on the raw file contents
+			// Have to serialize contents again first, since we need raw file contents for the intial text model content
 			let manager = await this.getOrCreateSerializationManager(providerId, uri);
 			serializedContent = await manager.contentManager.serializeNotebook(contents);
 		}
 
 		let options: INotebookShowOptions = {
 			providerId: providerId,
-			initialContent: serializedContent
+			initialContent: contents,
```

Comment:
What's going on here - why are we providing the same content twice (one serialized and one not)?

Reply:
The string content is used to initialize the underlying text model, since it's based on raw file contents. The JSON notebook contents are used to preserve any metadata or other info that would be lost during deserialization. This is a workaround for the fact that we pass around the notebook content as a string in a lot of places, but not all file representations (like DIB) have in-file metadata. It only exists while open in ADS.